autorest/codegen/serializers/__init__.py

```diff
@@ -491,3 +507,158 @@ def _serialize_and_write_metadata(
         self._autorestapi.write_file(
             namespace_path / Path("_metadata.json"), metadata_serializer.serialize()
         )
+
+    # find root folder where "setup.py" is
+    def _package_root_folder(self, namespace_path: Path) -> Path:
+        depth = self.code_model.namespace.count(".") + 1
+        return namespace_path / Path("../" * depth)
+
+    def _prepare_sample_render_param(self) -> Dict[Any, Any]:
```

Comment:
can you move this code to a sample serializer?

I think this code would look cleaner in another file. Some of this code also looks like it should live in the sample's jinja template